staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -69,13 +73,28 @@ type FeatureSpec struct {
 	Default bool
 	// LockToDefault indicates that the feature is locked to its default and cannot be changed
 	LockToDefault bool
-	// PreRelease indicates the maturity level of the feature
+	// PreRelease indicates the current maturity level of the feature
 	PreRelease prerelease
+	// Version indicates the earliest version from which this FeatureSpec is valid.
+	// If multiple FeatureSpecs exist for a Feature, the one with the highest version that is less
+	// than or equal to the effective version of the component is used.
+	Version *version.Version
 }
 
+type VersionedSpecs []FeatureSpec
+
+func (g VersionedSpecs) Len() int { return len(g) }
+func (g VersionedSpecs) Less(i, j int) bool {
+	return g[i].Version.LessThan(g[j].Version)
+}
+func (g VersionedSpecs) Swap(i, j int) { g[i], g[j] = g[j], g[i] }
+
+type PromotionVersionMapping map[prerelease]string
+
 type prerelease string
 
 const (
+	PreAlpha = prerelease("PRE-ALPHA")
```

Comment:
ok, I found below it is used for features didn't exist at that time

Reply:
With this change, the feature gate system will sometimes need to reason about features that are "not even alpha" the the version being emulated.  An example of this would be:

- kube-apiserver feature X is introduced as Alpha in 1.32
- The kube-apiserver (binary version `1.32`) is started with `--emulation-version=1.31`
- Because the binary is aware of feature X, it will need to assign it a stability level, but it is not even alpha yet, so it needs an identifier in code.

So this is not a new stability level, just a way to track stability levels within components in the presence of emulation versions.  The only way we plan for this to be exposed to users if if they try to enable a feature while also setting emulation version, resulting in an error like `"cannot set feature gate %v to %v, feature is PreAlpha at emulated version %s"`.